lib/spack/spack/util/compression.py

```diff
@@ -11,7 +11,7 @@
 # Supported archive extensions.
 PRE_EXTS   = ["tar", "TAR"]
 EXTS       = ["gz", "bz2", "xz", "Z"]
-NOTAR_EXTS = ["zip", "tgz", "tbz2", "txz"]
+NOTAR_EXTS = ["zip", "tgz", "tbz", "tbz2", "txz"]
```

Comment:
LGTM, just one question: Should we update `re.match(r'bz2', extension)` below to be `re.match(r'bz2?', extension)` so they are handled the same? On the other hand, xz also seems to be handled differently already.

Reply:
I've never seen a `.bz` file.

Reply:
My understanding was that this would also match .tbz2 files (and potentially .tbz). Probably doesn't matter anyway.

Reply:
Nope, we search for `.bz2`, not `bz2`.